client/app/components/modals/base-modal-controller.js

```diff
@@ -0,0 +1,42 @@
+(function() {
+  'use strict';
+
+  angular.module('app.components')
+    .controller('BaseModalController', BaseModalController);
+
+  /** @ngInject */
+  function BaseModalController($modalInstance, $state, CollectionsApi, EventNotifications) {
+    var base = this;
+    base.cancel = cancel;
+    base.reset = reset;
+    base.save = save;
+
+    function cancel() {
+      $modalInstance.dismiss();
+    }
+
+    function reset(event) {
+      angular.copy(event.original, this.modalData);
+    }
+
+    function save() {
+      var self = this;
```

Comment:
Same here I don't like it at all but I could not figure out another way, I'll tinker around a bit and see if I can come up with a cleaner solution.